nucliadb/src/migrations/0028_extracted_vectors_reference.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""Migration #28
+
+Add a key to each vectorset to know how to build the storage key for extracted vectors
+"""
+
+import logging
+
+from nucliadb.common import datamanagers
+from nucliadb.migrator.context import ExecutionContext
+from nucliadb_protos import knowledgebox_pb2
+
+logger = logging.getLogger(__name__)
+
+
+async def migrate(context: ExecutionContext) -> None: ...
+
+
+async def migrate_kb(context: ExecutionContext, kbid: str) -> None:
+    async with datamanagers.with_rw_transaction() as txn:
+        vectorsets = [vs async for (_vid, vs) in datamanagers.vectorsets.iter(txn, kbid=kbid)]
+
+        if len(vectorsets) == 0:
+            # should never happen, everyone should have at least one
+            logger.warning(f"KB has no vectorsets!", extra={"kbid": kbid})
```

Comment:
nit: `continue` here, otherwise it commits at the end.

Should we fail hard if this happens though? This is what you already checked in prod right?

Reply:
Ups, adding it!
No, I think is fine with a log. I'll check it when it runs just in case but yeah, is what I validated so nobody should have a KB like this